LanguageExt.Core/DataTypes/Seq/Seq.cs

```diff
@@ -808,20 +808,14 @@ public override int GetHashCode()
         /// <summary>
         /// Equality test
         /// </summary>
-        public bool Equals(Seq<A> rhs)
-        {
-            var x = this;
-            var y = rhs;
-
-            if (x.seq == null && y.seq == null &&
-                (x.Count != y.Count ||
-                 x.GetHashCode() != y.GetHashCode()))
-            {
-                return false;
-            }
-
-            return Enumerable.SequenceEqual(this, rhs);
-        }
+        public bool Equals(Seq<A> rhs) =>
+            !ReferenceEquals(rhs, null) &&
+            this.seq == null && rhs.seq == null &&
```

Comment:
Is it possible for `ReferenceEquals(this.seq, rhs.seq)` to be true (e.g. could we ever have two different `Seq<>` that have the same inner `seq`)?  If so, would that be worth inserting between lines 812 and 813?